lib/AppInfo/Application.php

```diff
@@ -55,5 +58,27 @@ public function __construct() {
 				$Container->query('ImageDimensionMapper')
 			);
 		});
+
+		$container->registerService('RetrieveMetadata', function(IContainer $Container) {
+			return new RetrieveMetadata(
+				$Container->query('ServerContainer')->getDb()
+			);
+		});
+
+		$container->registerService('MetadataController', function(IContainer $container){
+			return new MetadataController(
+				$container->query('AppName'),
+				$container->query('Request'),
+				$container->query('RetrieveMetadata')
+			);
+		});
+
+		$container->registerService('MetadataApiController', function(IContainer $container){
+			return new MetadataApiController(
+				$container->query('AppName'),
+				$container->query('Request'),
+				$container->query('RetrieveMetadata')
+			);
+		});
```

Comment:
Could you explain why you needed to explicitly create the services? Did you see that the metadata wasn't stored during setup without this?